apps/sms/views/shared/js/utils.js

```diff
@@ -711,6 +711,57 @@
     },
 
     /**
+     * Returns a rate limited function, useful to reduce the number of function
+     * calls when listening to user input (keypress) or browser events.
+     * @params {function} func - The function that should be rate-limited
+     * @params {number=300} delay - Min amount of time (ms) between two calls
+     * @param {Boolean=false} preventfirstCall - Prevent first synchronous call 
+     * when true.
+     * @param {Boolean=false} preventLastCall - Prevent last trailing call
+     * when true.
+     * @returns {function} - The rate limited function that will call the
+     *  original function
+     */
+    throttle: function(func,
+                       delay = 300, 
+                       preventfirstCall = false, 
```

Comment:
Thanks for the reminder, we can use destructuring assignement actually since 2.5 must binds with gecko version later then 41(should be 43) so you can apply destructuring assignement here if you want.